source/developer/webhooks-incoming.rst

```diff
@@ -3,128 +3,213 @@
-  b. To get the message posted into Mattermost, your integration will need to create an HTTP POST request that will submit to the incoming webhook URL you created before. The body of the request must have a *payload* that contains a JSON object that specifies a *text* parameter. For example, ``payload={"text": "Hello, this is some text."}``` is a valid body for a request.
-  c. Set up your integration running on Heroku, an AWS server or a server of your own to start sending real-time updates to Mattermost channels.
+For example, to create a message with a heading, and an italicized text on the next line, use the following payload. 
 
-**Additional Notes:**
+  .. code-block::
 
-1. For the HTTP request body, if `Content-Type` is specified as `application/json` in the headers of the HTTP request then the body of the request can be direct JSON. For example, ``{"text": "Hello, this is some text."}``
+    payload={"text": "# This is a heading\n_This text is italicized._"}
 
-2. You can override the channel specified in the webhook definition by specifying a `channel` parameter in your payload. For example, you might have a single webhook created for *Town Square*, but you can use ``payload={"channel": "off-topic", "text": "Hello, this is some text."}`` to send a message to the *Off-Topic* channel using the same webhook URL. If an *@* symbol followed by a username is specified, then the message will be sent to that user's direct message channel.
+.. image:: ../images/incoming_webhooks_markdown_formatting.png
 
-3. In addition, with **Enable integrations to override usernames** turned on,  you can also override the username the message posts as by providing a *username* parameter in your JSON payload. For example, you might want your message looking like it came from a robot so you can use ``payload={"username": "robot", "text": "Hello, this is some text."}`` to change the username of the post to "robot". Note, to combat any malicious users from trying to use this to perform `phishing attacks <https://en.wikipedia.org/wiki/Phishing>`_ a *BOT* indicator appears next to posts coming from webhooks.
+Messages with advanced formatting can be created by including an :doc:`attachment array <message-attachments>` in the JSON payload.
 
-4. With **Enable integrations to override profile picture icons** turned on, you can similarly change the icon the message posts with by providing a link to an image in the *icon_url* parameter of your payload. For example, ``payload={"icon_url": "http://example.com/somecoolimage.jpg", "text": "Hello, this is some text."}`` will post using whatever image is located at *http://example.com/somecoolimage.jpg* as the icon for the post.
+Complete incoming webhook example
```

Comment:
Wondering if we should show a complete example at the top so information is available at a glance